Remotion/Data/DomainObjects/Tracing/TracingDbConnection.cs

```diff
@@ -16,137 +16,177 @@
-      _connectionID = Guid.NewGuid();
+    public override Task OpenAsync (CancellationToken cancellationToken)
+    {
+      return _connection.OpenAsync(cancellationToken);
     }
 
-    public IDbConnection WrappedInstance
+    public new TracingDbTransaction BeginTransaction ()
     {
-      get { return _connection; }
+      return (TracingDbTransaction)base.BeginTransaction();
     }
 
-    public Guid ConnectionID
+    public new TracingDbTransaction BeginTransaction (IsolationLevel isolationLevel)
     {
-      get { return _connectionID; }
+      return (TracingDbTransaction)base.BeginTransaction(isolationLevel);
     }
 
-    public IPersistenceExtension PersistenceExtension
+    protected override TracingDbTransaction BeginDbTransaction (IsolationLevel isolationLevel)
     {
-      get { return _persistenceExtension; }
+      var transaction = _connection.BeginTransaction(isolationLevel);
+      PersistenceExtension.TransactionBegan(_connectionID, transaction.IsolationLevel);
+      return CreateTracingTransaction(transaction);
+    }
+
+    public new TracingDbCommand CreateCommand ()
+    {
+      return (TracingDbCommand)base.CreateCommand();
+    }
+
+    protected override TracingDbCommand CreateDbCommand ()
+    {
+      return CreateTracingCommand(_connection.CreateCommand());
     }
 
-    public void Open ()
+    public override void Open ()
     {
       _connection.Open();
       PersistenceExtension.ConnectionOpened(_connectionID);
     }
 
-    public void Close ()
+    public override void Close ()
     {
       _connection.Close();
 
       TraceConnectionClosed();
     }
 
-    public void Dispose ()
+    public override async Task CloseAsync ()
     {
-      _connection.Dispose();
-
+      await _connection.CloseAsync();
       TraceConnectionClosed();
     }
 
-    public TracingDbTransaction BeginTransaction ()
+    protected override void Dispose (bool disposing)
     {
-      var transaction = _connection.BeginTransaction();
-      PersistenceExtension.TransactionBegan(_connectionID, transaction.IsolationLevel);
-      return CreateTracingTransaction(transaction);
-    }
+      if (disposing)
+      {
+        _connection.Dispose();
 
-    public TracingDbTransaction BeginTransaction (IsolationLevel isolationLevel)
-    {
-      var transaction = _connection.BeginTransaction(isolationLevel);
-      PersistenceExtension.TransactionBegan(_connectionID, isolationLevel);
-      return CreateTracingTransaction(transaction);
+        TraceConnectionClosed();
+      }
     }
 
-    public TracingDbCommand CreateCommand ()
+    public override async ValueTask DisposeAsync ()
     {
-      return CreateTracingCommand(_connection.CreateCommand());
+      await _connection.DisposeAsync();
+      TraceConnectionClosed();
     }
 
-    private TracingDbTransaction CreateTracingTransaction (IDbTransaction transaction)
+
+    private TracingDbTransaction CreateTracingTransaction (DbTransaction transaction)
```

Comment:
```suggestion
    private TracingDbTransaction CreateTracingTransaction (DbTransaction transaction)
```
There was an extra blank line